### PR TITLE
Added variable parameters for os & architectures [semver:minor]

### PR DESCRIPTION
**Pull request by punkdata**
Now that we support multiple Os and cpu architectures baking in some flexibility in what tf clients to d/l and install is critical. I have created some new parameters that enable os + arch value assignment.

### CODE REVIEW

**Comment by punkdata** [comment]
@KyleTryon @gmemstr pls review this change for the terraform orb thx

File: src/commands/install.yml [hunk 1]
```diff
@@ -6,13 +6,21 @@ parameters:
     type: "string"
     description: "Specify version of terraform to install."
     default: "0.13.5"
+  os:
+    type: string
```
Comment (KyleTryon) [inline src/commands/install.yml]: If we have a set of known values, let's use the enum type.
https://circleci.com/docs/2.0/reusing-config/#enum

File: src/commands/install.yml [hunk 2]
```diff
@@ -6,13 +6,21 @@ parameters:
...
+    description: "specify the operating system version to install [linux, darwin, windows]"
+    default: "linux"
+  arch:
+    type: string
```
Comment (KyleTryon) [inline src/commands/install.yml]: Same as above.

**Review by KyleTryon — CHANGES_REQUESTED** [review]

File: src/commands/install.yml (hunk 2 above)
Comment (punkdata) [reply src/commands/install.yml]: @KyleTryon I like this approach thx for calling out  👍🏽 I'll make the change

Comment (punkdata) [reply src/commands/install.yml]: @KyleTryon I adjusted the params based on your suggestion. thx

File: src/commands/install.yml (hunk 1 above)
Comment (punkdata) [reply src/commands/install.yml]: done

**Review by punkdata — COMMENTED** [review]
done

File: src/commands/install.yml [hunk 3]
```diff
@@ -6,13 +6,23 @@ parameters:
...
     description: "Specify version of terraform to install."
     default: "0.13.5"
+  os:
+    type: enum
+    description: "specify the operating system version to install. Must one of these values: linux, darwin"
+    default: "linux"
+    enum: ["linux", "darwin"]
+  arch:
+    type: enum
+    description: "specify the platform architecture to install. Must one of these values: amd64, arm64"
```
Comment (KyleTryon) [inline src/commands/install.yml]: must be one

File: src/commands/install.yml [hunk 4]
```diff
@@ -6,13 +6,23 @@ parameters:
     type: "string"
     description: "Specify version of terraform to install."
     default: "0.13.5"
+  os:
+    type: enum
+    description: "specify the operating system version to install. Must one of these values: linux, darwin"
```
Comment (KyleTryon) [inline src/commands/install.yml]: must be one

**Review by KyleTryon — CHANGES_REQUESTED** [review]

File: src/commands/install.yml (hunk 3 above)
Comment (punkdata) [reply src/commands/install.yml]: fixed

File: src/commands/install.yml (hunk 4 above)
Comment (punkdata) [reply src/commands/install.yml]: fixed

**Review by punkdata — COMMENTED** [review]
fixed typos

**Review by KyleTryon — APPROVED** [review]

**Comment by punkdata** [comment]
Gracias @KyleTryon much appreciated 👍🏽

**Pull request merged**